test/dataset_collection/test_dataset_collection.py

```diff
@@ -18,165 +18,156 @@
 class TestDatasetCollection(TestCase):
     current_directory = Path(__file__).absolute().parent
 
-    def test_determine_type_without_return_without_arguments(self):
+    def setUp(self):
         filepath = self.current_directory / "Example.java"
-        ast = AST.build_from_javalang(build_ast(filepath))
+        self.ast = AST.build_from_javalang(build_ast(filepath))
+        self.temp_filename = self.current_directory / 'temp.java'
+
+    def tearDown(self):
+        if self.temp_filename.exists():
```

Comment:
is it better to create filename as parameter, not and the field of the class
def tearDown(self, filename: Path):

Reply:
why is it better?